### PR TITLE
fix(restore): [FOR-424] fix restoration for shared forms

### PR DESCRIPTION
**Pull request by qperie**
## Describe your changes
When we restore a form shared to someone, it's restored for the person who restored it but not for the other ones.
A new API endpoint was created specically for it.

## Issue ticket number and link
FOR-424 : https://jira.support-ent.fr/browse/FOR-424

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [x] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: common/src/main/resources/ts/services/FormService.ts [hunk 1]
```diff
@@ -119,11 +119,9 @@ export const formService: FormService = {
         }
     },
 
-    async restore(form: Form, destinationFolderId: number) : Promise<any> {
+    async restore(forms: Form[]) : Promise<any> {
         try {
-            form.archived = false;
-            await this.move([form.id], destinationFolderId);
-            return await this.update(form);
+            return DataUtils.getData(await http.put(`/formulaire/forms/restore`, forms.map(f => f.id)));
```
Comment (BastienBenard) [inline common/src/main/resources/ts/services/FormService.ts:124]: typage

File: formulaire/src/main/java/fr/openent/formulaire/controllers/FormController.java [hunk 2]
```diff
@@ -831,6 +831,81 @@ else if (relFormFolderEvt.right().getValue().isEmpty()) {
...
+    public void restore(HttpServerRequest request) {
+        UserUtils.getUserInfos(eb, request, user -> {
+            if (user == null) {
+                String message = "[Formulaire@restoreForms] User not found in session.";
+                log.error(message);
+                unauthorized(request, message);
+                return;
+            }
+
+            RequestUtils.bodyToJsonArray(request, formIds -> {
```
Comment (BastienBenard) [inline formulaire/src/main/java/fr/openent/formulaire/controllers/FormController.java:847]: Tu peux utiliser bodyToJson(final HttpServerRequest request, final String schema, final Handler<JsonObject> handler)
exemple
RequestUtils.bodyToJson(request, pathPrefix + "alertDelete", body -> {}

https://github.com/OPEN-ENT-NG/presences/blob/MA-770/presences/src/main/resources/jsonschema/alertDelete.json

**Review by BastienBenard — APPROVED** [review]

File: formulaire/src/main/java/fr/openent/formulaire/controllers/FormController.java [hunk 3]
```diff
@@ -831,6 +831,81 @@ else if (relFormFolderEvt.right().getValue().isEmpty()) {
...
+                                renderInternalError(request, formsEvt);
+                                return;
+                            }
+
+                            JsonArray forms = formsEvt.right().getValue();
+                            for (int i = 0; i < forms.size(); i++) {
+                                forms.getJsonObject(i).put(ARCHIVED, false);
+                            }
+
+                            formService.updateMultiple(forms, arrayResponseHandler(request));
```
[the remark this answers is not in the document]
Comment (qperie) [reply formulaire/src/main/java/fr/openent/formulaire/controllers/FormController.java:901]: J'ai commencé des changements mais ça en entraîne d'autres et y'a quelques points que je voudrais éclaircir d'abord. Je met donc ça de coté pour un autre ticket mais très intéressant les compose 👍

File: formulaire/src/main/java/fr/openent/formulaire/controllers/FormController.java (hunk 2 above)
Comment (qperie) [reply formulaire/src/main/java/fr/openent/formulaire/controllers/FormController.java:847]: Comme pour les magics string récemment, je ferai une passe globale pour les schema json

**Pull request merged**